app/scripts/modules/core/src/presentation/modal/Modal.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import { CSSTransition } from 'react-transition-group';
+
+const { useMemo } = React;
+
+import { useEventListener, useContainerClassNames, useLatestCallback } from 'core/presentation';
+
+import { ModalContext } from './ModalContext';
+import * as styles from './Modal.module.css';
+
+export interface IModalProps {
+  isOpen: boolean;
+  onRequestClose?: () => any;
+  children?: React.ReactNode;
+}
+
+export const Modal = ({ onRequestClose, isOpen, children }: IModalProps) => {
+  useContainerClassNames(isOpen ? [styles.backdropBlurEffect] : []);
+
+  const keydownCallback = ({ keyCode }: KeyboardEvent) => {
+    if (keyCode === 27) {
+      onRequestClose();
```

Comment:
should this use `memoizedRequestClose`?

Reply:
it's functionally equivalent to use either one here, because `useEventListener` also uses `useLatestCallback` under the hood so you can write event callbacks without worrying about closure freshness. Perhaps it's a bit pedantic, but I've generally been making an effort to only use memoized/ref'd versions of things in places where it's functionally necessary as a way to communicate meaning/significance to the reader.